src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1409,68 +1481,101 @@ impl Cage {
                             }
```

Comment:
Why is this not an error? does the read_dat function return a None type value only when there is nothing to be read?

Reply:
I referenced the man-pages for `read_syscall` and they have mentioned that if read is not successful, then we return 0 as the default value in the response, representing no bytes were read from the file. 
https://man7.org/linux/man-pages/man2/read.2.html

Reply:
Updated both the code references as per the discussion.